localstack-core/localstack/services/events/provider.py

```diff
@@ -1132,7 +1130,7 @@ def func(*args, **kwargs):
 
                 target_sender = self._target_sender_store[target["Arn"]]
                 try:
-                    target_sender.process_event(event, context)
+                    target_sender.process_event(event.copy())
```

Comment:
Can we add a comment in the code clarifying why and wherever `.copy()` is needed?
If EventStudio depends on specific behavior, we should explicitly add comments so that removing the particular code doesn't break It.